lib/webflow/client.rb

```diff
@@ -21,85 +21,104 @@ def remaining
+      custom_domains ||= domains(site_id).map { |domain| domain[:name] }
+      post("/sites/#{site_id}/publish", { publishToWebflowSubdomain: true, customDomains: custom_domains })
     end
 
     def collections(site_id)
-      get("/sites/#{site_id}/collections")
+      get("/sites/#{site_id}/collections").fetch(:collections)
     end
 
     def collection(collection_id)
       get("/collections/#{collection_id}")
     end
 
-
     # https://developers.webflow.com/?javascript#get-all-items-for-a-collection
     # returns json object with data to help paginate collection
     #
     # {
-    #   items:  your list of items returned,
-    #   count:  number of items returned,
-    #   limit:  the limit specified in the request (default: 100),
-    #   offset: the offset specified for pagination (default: 0),
-    #   total:  total # of items in the collection
+    #   items: [] your list of items returned,
+    #   pagination: {
+    #     limit:  the limit specified in the request (default: 100),
+    #     offset: the offset specified for pagination (default: 0),
+    #     total:  total # of items in the collection
+    #   }
     # }
     #
     # page starts at 1
     def paginate_items(collection_id, per_page: 100, page: 1)
       get("/collections/#{collection_id}/items", params: { limit: per_page, offset: per_page * (page - 1) })
     end
 
-
-    def items(collection_id, limit: 100)
+    def items(collection_id, limit: 100) # rubocop:disable Metrics/MethodLength
       fetched_items = []
       num_pages     = (limit.to_f / 100.0).ceil
-      per_page      = limit > 100 ? 100 : limit
+      per_page      = [limit, 100].min
 
       num_pages.times do |i|
-        resp = paginate_items(collection_id, per_page: per_page, page: i+1)
-        items = resp['items']
-        yield(items) if block_given?
-        fetched_items += items
-        limit -= resp['count']
-        break if limit <= 0 || resp['total'] <= fetched_items.length
+        response = paginate_items(collection_id, per_page: per_page, page: i + 1)
+        items = response.fetch(:items)
+
+        if block_given?
+          yield(items)
+        else
+          fetched_items += items
+        end
       end
 
       fetched_items
     end
 
     def item(collection_id, item_id)
-      json = get("/collections/#{collection_id}/items/#{item_id}")
-      json['items'].first
+      get("/collections/#{collection_id}/items/#{item_id}")
+    end
+
+    def create_item(collection_id, data, is_archived: false, is_draft: false, publish: false)
+      result = post("/collections/#{collection_id}/items",
+                    { isArchived: is_archived, isDraft: is_draft, fieldData: data })
+      return result unless publish
+
+      publish_item(collection_id, result.fetch(:id))
+      result
+    end
+
+    def update_item(collection_id, item_id, data, is_archived: false, is_draft: false, publish: false) # rubocop:disable Metrics/ParameterLists
```

Comment:
increase the limit or disable this rule instead please.

Reply:
Hmmm, this rule has those limitations for a reason.
https://docs.rubocop.org/rubocop/cops_metrics.html#metricsparameterlists

I believe the default limit of 5 is a good limit, maybe even too generous/high. It was only disabled here in one place. I find that to be better than increasing the limit globally to 6, which makes this rule's function almost useless, and allows developers to easily add one more argument, as there's no "warning" about having too many arguments.
https://thoughtbot.com/blog/sandi-metz-rules-for-developers#four-method-arguments

I think disabling rules inline is fine. It stands right there in the code, in your face, so that you can see it and try to figure out a way to refactor your code so that you can remove the inline disable.

But if you really have strong opinion, perhaps we could add `CountKeywordArgs: false` and we could remove the inline disable?

Reply:
i dislike 90% of the preconfigured rubocop rules, so please ignore me if you feel different about it. i just really hate cluttering the code for a tool. that should generally be forbidden.